### PR TITLE
HLRC: Add ML get influencers API

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Relates #29827

### CODE REVIEW

**Review by benwtrent — COMMENTED** [review]
Some copy/paste and find/replace errors are the biggest things.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetInfluencersRequest.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    public String getEnd() {
+        return end;
+    }
+
+    /**
+     * Sets the value of "end" which is a timestamp.
+     * Only influencers whose timestamp is before the "end" value will be returned.
+     * @param end value of "end" to be set
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetInfluencersRequest.java]: I have also taken the opportunity to fix this in other get result request objects.

**Comment by dimitris-athanasiou** [comment]
@benwtrent All good points! Pushed a commit addressing all of them.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetInfluencersRequest.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        PARSER.declareBoolean(GetInfluencersRequest::setExcludeInterim, EXCLUDE_INTERIM);
+        PARSER.declareStringOrNull(GetInfluencersRequest::setStart, START);
+        PARSER.declareStringOrNull(GetInfluencersRequest::setEnd, END);
+        PARSER.declareObject(GetInfluencersRequest::setPageParams, PageParams.PARSER, PageParams.PAGE);
+        PARSER.declareDouble(GetInfluencersRequest::setInfluencerScore, INFLUENCER_SCORE);
+        PARSER.declareString(GetInfluencersRequest::setSort, SORT);
+        PARSER.declareBoolean(GetInfluencersRequest::setDescending, DESCENDING);
+    }
+
+    private String jobId;
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetInfluencersRequest.java]: should be `final`

**Review by benwtrent — APPROVED** [review]

**Pull request merged**